modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module

```diff
@@ -56,20 +61,162 @@ function oe_editorial_corporate_workflow_menu_local_tasks_alter(&$data): void {
 /**
  * Implements hook_form_FORM_ID_alter().
  *
- * Force the Draft state on node edit forms.
+ * For the Node form to force Draft state except when the node is published.
  */
 function oe_editorial_corporate_workflow_form_node_form_alter(&$form, FormStateInterface $form_state, $form_id) {
   /** @var \Drupal\node\NodeInterface $node */
-  $node = $form_state->getFormObject()->getEntity();
-  if ($node instanceof NodeInterface && !$node->isNew() && _oe_editorial_corporate_workflow_is_last_revision_published($node)) {
+  $entity = $form_state->getFormObject()->getEntity();
+  if (!$entity instanceof NodeInterface) {
+    return;
+  }
+  /** @var \Drupal\content_moderation\ModerationInformationInterface $moderation_info */
+  $moderation_info = \Drupal::service('content_moderation.moderation_information');
+  /** @var \Drupal\workflows\WorkflowInterface $workflow */
+  $workflow = $moderation_info->getWorkflowForEntity($entity);
+  if (!$workflow || $workflow->id() !== 'oe_corporate_workflow') {
+    return;
+  }
+
+  $form_options = &$form['moderation_state']['widget'][0]['state']['#options'];
+  if (empty($form_options)) {
     return;
   }
 
-  if (isset($form['moderation_state']['widget'][0]['state']['#options']['draft'])) {
-    $form['moderation_state']['widget'][0]['state']['#options'] = [
-      'draft' => $form['moderation_state']['widget'][0]['state']['#options']['draft'],
-    ];
+  // Enforce Draft state in forms.
+  /** @var \Drupal\workflows\WorkflowInterface $workflow */
+  $workflow = $moderation_info->getWorkflowForEntity($entity);
+  $options = [
+    'draft' => $workflow->getTypePlugin()->getState('draft')->label(),
+  ];
+
+  // Remove access because Draft is the only option.
+  $form['moderation_state']['widget'][0]['state']['#access'] = FALSE;
+  $form['moderation_state']['widget'][0]['state']['#default_value'] = 'draft';
+
+  // However, if the node is in Published state, we might need to append
+  // a few more options.
+  if (_oe_editorial_corporate_workflow_is_last_revision_published($entity)) {
+    // 'Expired' and 'Archived' are only available through 'Edit form'
+    // and they always show up together.
+    if (isset($form_options['expired'])) {
+      $options['expired'] = $form_options['expired'];
+      $options['archived'] = $form_options['archived'];
+      $form['moderation_state']['widget'][0]['state']['#access'] = TRUE;
+    }
   }
+
+  $form_options = $options;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * For the content moderation block form to make sure next state is selected
+ * in the widget.
+ */
+function oe_editorial_corporate_workflow_form_content_moderation_entity_moderation_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if (empty($form['new_state']['#options'])) {
+    return;
+  }
+
```

Comment:
Misses the check to only alter in case of our corporate workflow.
if (!$workflow || $workflow->id() !== 'oe_corporate_workflow') {